### PR TITLE
Refactor tests and scripts

### PR DESCRIPTION
**Pull request by ahuth**
This PR makes a couple refactors:
- `npm run test` only runs unit tests now
- Unit tests are now co-located with their source files
- The TS build now excludes test files
- Configures the eslint Jest plugin to actually run its recommended rules
- Always build the demo app before running integration tests

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
         run: npm run lint
 
       - name: Unit Tests
-        run: npm run test:unit
+        run: npm run test
```
Comment (jeremiah-clothier) [inline .github/workflows/tests.yml:28]: Do the `tests/unit` still get run as part of ci?

**Review by jeremiah-clothier — APPROVED** [review]
Non-blocking question

File: .github/workflows/tests.yml (hunk 1 above)
Comment (jeremiah-clothier) [reply .github/workflows/tests.yml:28]: edit: nvm I see you renamed all the files that were in that directory

File: package.json [hunk 2]
```diff
@@ -36,8 +36,8 @@
     "lint": "eslint --ext=js,jsx,ts,tsx .",
     "prebuild": "npm run clean",
     "prepublishOnly": "npm run build",
-    "pretest:integration": "npm run build",
-    "test": "npm run test:unit && npm run test:integration",
+    "pretest:integration": "npm run build && npm run demo:build",
+    "test": "jest src",
     "test:integration": "npm run test:integrationOnly",
     "test:integrationOnly": "jest tests/integration",
     "test:unit": "jest tests/unit",
```
Comment (jeremiah-clothier) [inline package.json:43]:
```suggestion
```

**Comment by ahuth** [comment]
Thanks

**Pull request merged**